File: codigo-fonte/java/calculo-fatorial/CalcularFactorial.java
import java.math.BigInteger;

public class CalcularFactorial {

	public BigInteger factorial(BigInteger f, BigInteger n) {
		if (n.compareTo(new BigInteger("1")) == 0)
			return f;
		else
			return factorial(f.multiply(n), n.subtract(new BigInteger("1")));
	}

	public static void main(String[] args) {
		double tempoInicial = System.nanoTime();
		
		CalcularFactorial c = new CalcularFactorial();
		BigInteger resultado = c.factorial(new BigInteger("1"),  new BigInteger("50"));
		
		double tempoFinal = System.nanoTime();

		System.out.println("Resultado: " + resultado);
		System.out.println("Tempo (ms): " + ((tempoFinal - tempoInicial) / 1000000));
	}
}

File: codigo-fonte/scala/calculo-fatorial/CalcularFactorial.scala
object CalcularFactorial extends App {

  val tempoInicial: Double = System.nanoTime

  def factorial(f: BigInt, n: BigInt): BigInt = {
    if (n == 1) f else factorial((f * n), (n - 1))
  }

  val resultado = factorial(1, 50)
  val tempoFinal: Double = System.nanoTime

  println(s"Resultado: $resultado")
  println(s"Tempo (ms): ${((tempoFinal - tempoInicial) / 1000000)}")

}
